### PR TITLE
Add github actions 

### PR DESCRIPTION
**Pull request by mllemango**
### WHY are these changes introduced?
This library is running on travis-ci.org, [which will be going into read-only mode at the end of the year](https://docs.travis-ci.com/user/migrate/open-source-repository-migration/#frequently-asked-questions)
> we are announcing that travis-ci.org will be officially closed down completely no later than December 31st, 2020

Taking this opportunity to move away from travis-ci all together, and onto Github Actions. 
It is my understanding that we aren't billed for action usage on public repos.

### WHAT is this pull request doing?
Covering the CI pipeline with Github Actions. 

### Checklist

- [ ] I have updated the CHANGELOG (if applicable)
- [ ] I have followed the [Shopify Python](https://github.com/Shopify/shopify_python) guide

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  push:
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Python ${{ matrix.version }}
+    strategy: 
+      matrix:
+        version: [2.7, 3.4, 3.5, 3.6]
```
Comment (mllemango) [inline .github/workflows/build.yml:13]: Wondering if we should pop in more versions to test?

Comment (andyw8) [reply .github/workflows/build.yml:13]: We'll keep 2.7, and for the 3 series I think we should cover the latest release, and the oldest supported, so 3.6 and 3.9. I don't think we need to cover the in between ones.

**Comment by andyw8** [comment]
Let's also remove the `.travis.yml`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.version }}
+      - name: Install Dependencies
+        run: | 
+          python -m pip install --upgrade pip
+          pip install flake8 pytest
+          pip install mock
```
Comment (andyw8) [inline .github/workflows/build.yml:25]: Any idea why the travis config didn't need these additional steps?

Comment (mllemango) [reply .github/workflows/build.yml:25]: Really not sure! :/

**Review by andyw8 — APPROVED** [review]

**Pull request merged**